### PR TITLE
CCS-11: Create Route Handler for Update Functionality

### PR DESCRIPTION
**Pull request by jmusone**
Added update clothing handler and added service config.

### CODE REVIEW

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    private final ClothingService service;
+
+    public UpdateClothingHandler(ClothingService service) {
+        this.service = service;
+    }
+
+    @Override
+    public Mono<ServerResponse> handle(ServerRequest serverRequest) {
+        return serverRequest.bodyToMono(Request.class)
+                .flatMap(new Function<Request, Mono<? extends Clothing>>() {
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java:29]: This should be a map() not flatMap()

Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java]: Make this a lambda and remove any casting

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                                .condition(request.getCondition())
+                                .status(Item.Status.PRESENT)
+                                .brand(request.getBrand())
+                                .color(request.getColor())
+                                .type(request.getType())
+                                .gender(request.getGender())
+                                .size(request.getSize())
+                                .build());
+                    }
+                }).flatMap(new Function<Clothing, Mono<? extends ServerResponse>>() {
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java]: Same as above

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                    @Override
+                    public Mono<? extends ServerResponse> apply(Clothing clothing) {
+                        return ServerResponse.ok()
+                                .body(BodyInserters.fromValue(Response.from(clothing)));
+                    }
+                })
+                .onErrorResume(ex -> ServerResponse.badRequest().build());
```
Comment (hitchan) [inline InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java]: Theres generally two error scenarios that should be covered.
1. onErrorResume() will catch every error downstream from itself, most of these are going to be internal server errors.
2. Under onErrorResume() you could put a switchIfEmpty() operator to catch the scenario where the service cant find the entity with the provided ID. You would want to return a 404 not found in that case.

**Review by hitchan — CHANGES_REQUESTED** [review]
Just a couple of things to address but looks good in general.

**Review by RRDarius — APPROVED** [review]

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java (hunk 1 above)
Comment (jmusone) [reply InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java]: Updated

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java (hunk 2 above)
Comment (jmusone) [reply InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java]: Updated

File: InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java (hunk 3 above)
Comment (jmusone) [reply InventoryManagement/src/main/java/com/ccs/inventorymanagement/route/UpdateClothingHandler.java]: Updated

**Review by hitchan — APPROVED** [review]

**Pull request merged**